src/System.Runtime/tests/System/CharTests.cs

```diff
@@ -116,19 +115,19 @@ public static void ConvertToUtf32_String_Int_Invalid()
             Assert.Throws<ArgumentOutOfRangeException>("index", () => char.ConvertToUtf32("", 0)); // Index >= string.Length
         }
 
-        [Fact]
-        public static void ConvertToUtf32_Char_Char()
+        public static IEnumerable<object[]> ConvertToUtf32_Char_Char_TestData()
         {
-            // TODO: add this as [InlineData] when #7166 is fixed
-            TestConvertToUtf32_Char_Char('\uD800', '\uDC00', 0x10000);
-            TestConvertToUtf32_Char_Char('\uD800', '\uDC00', 0x10000);
-            TestConvertToUtf32_Char_Char('\uD800', '\uDFFF', 0x103FF);
-            TestConvertToUtf32_Char_Char('\uDBBF', '\uDFFF', 0xFFFFF);
-            TestConvertToUtf32_Char_Char('\uDBFF', '\uDC00', 0x10FC00);
-            TestConvertToUtf32_Char_Char('\uDBFF', '\uDFFF', 0x10FFFF);
+            yield return new object[] { '\uD800', '\uDC00', 0x10000 };
+            yield return new object[] { '\uD800', '\uDC00', 0x10000 };
+            yield return new object[] { '\uD800', '\uDFFF', 0x103FF };
+            yield return new object[] { '\uDBBF', '\uDFFF', 0xFFFFF };
+            yield return new object[] { '\uDBFF', '\uDC00', 0x10FC00 };
+            yield return new object[] { '\uDBFF', '\uDFFF', 0x10FFFF };
         }
 
-        private static void TestConvertToUtf32_Char_Char(char highSurrogate, char lowSurrogate, int expected)
+        [Theory]
+        [MemberData(nameof(ConvertToUtf32_Char_Char_TestData))]
```

Comment:
Same question.  To me, MemberData makes sense when there's data that can't be represented in an attribute, when it's computed programmatically (e.g. via a loop), or when the data is going to be shared across multiple tests.  Otherwise, I prefer we use InlineData, as it keeps the data as close to the relevant test as possible, and does so with as little ceremony as possible.